jerry-core/jmem/jmem.h

```diff
@@ -223,6 +245,38 @@ void * JERRY_ATTR_PURE jmem_decompress_pointer (uintptr_t compressed_pointer);
     } \
   } while (false);
 
+/**
+ * Set value of pointer-tag value so that it will correspond
+ * to specified non_compressed_pointer along with tag
+ */
+#define JMEM_CP_SET_NON_NULL_POINTER_TAG(cp_value, pointer, tag) \
+  do \
+  { \
+    JERRY_ASSERT ((uintptr_t) tag < (uintptr_t) (JMEM_ALIGNMENT)); \
+    jmem_cpointer_tag_t compressed_ptr = jmem_compress_pointer (pointer); \
+    (cp_value) = (jmem_cpointer_tag_t) ((compressed_ptr << JMEM_TAG_SHIFT) | tag); \
+  } while (false);
+
+/**
+ * Extract value of pointer from specified pointer-tag value
+ */
+#define JMEM_CP_GET_NON_NULL_POINTER_FROM_POINTER_TAG(type, cp_value) \
+  ((type *) (jmem_decompress_pointer ((cp_value & ~JMEM_TAG_MASK) >> JMEM_TAG_SHIFT)))
+
+/**
+ * Get value of each tag from specified pointer-tag value
+ */
+#define JMEM_CP_GET_FIRST_TAG_FROM_POINTER_TAG(cp_value) (cp_value & JMEM_FIRST_TAG_MASK)
```

Comment:
I would call these BIT-s, because they are not represent something generic, only a bit. 
JMEM_CP_GET_FIRST_BIT_FROM_POINTER_TAG

The three bits together can be the tag.